dags/dag_content/tower_metrics_content.py

```diff
@@ -0,0 +1,411 @@
+        col_headers.append(item.get("name"))
+    # handle single-value queries
+    if len(col_headers) < 2:
+        final_dict = {query_name: list(response["records"][0][0].values())[0]}
+        return final_dict
+    # handle n-value queries
+    else:
+        data_dict = {}
+        for item in response["records"]:
+            key = list(item[0].values())[0]
+            sub_dict = {key: {}}
+            for i in [x for x in range(len(col_headers)) if x != 0]:
+                sub_dict[key].update({col_headers[i]: list(item[i].values())[0]})
+            data_dict.update(sub_dict)
+        final_dict = {query_name: data_dict}
+        return final_dict
+
+
+# creates RDS boto3 client inside task
+def create_rds_client(aws_creds):
+    rds = boto3.client(
+        "rds",
+        aws_access_key_id=aws_creds["AWS_ACCESS_KEY_ID"],
+        aws_secret_access_key=aws_creds["AWS_SECRET_ACCESS_KEY"],
+        region_name=AWS_REGION
+    )
+    return rds
+
+
+# creates RDS Data boto3 client inside task
+def create_rds_data_client(aws_creds):
+    rdsData = boto3.client(
+        "rds-data",
+        aws_access_key_id=aws_creds["AWS_ACCESS_KEY_ID"],
+        aws_secret_access_key=aws_creds["AWS_SECRET_ACCESS_KEY"],
+        region_name=AWS_REGION
+    )
+    return rdsData
+
+
+# creates Secrets Manager boto3 client inside task
+def create_secret_client(aws_creds):
+    secrets = boto3.client(
+        "secretsmanager",
+        aws_access_key_id=aws_creds["AWS_ACCESS_KEY_ID"],
+        aws_secret_access_key=aws_creds["AWS_SECRET_ACCESS_KEY"],
+        region_name=AWS_REGION
+    )
+    return secrets
+
+# create authenticated synapse session
+def create_synapse_session():
+    syn = synapseclient.Synapse()
+    syn.login(authToken=Variable.get("SYNAPSE_AUTH_TOKEN")) # TODO - this is currently Brad's synapse token
+    return syn
+
+#check if database process is complete - can be used for clone creation, modification and deletion
+def check_database_process_complete(client, waiter_type, db_name):
+    time.sleep(20) #allow process time to start before starting waiter
+    waiter = client.get_waiter(waiter_type)
+    waiter.wait(
+        DBClusterIdentifier = db_name
+    )
+
+#TASKS
+
+@task(multiple_outputs=True)
+def get_database_info(aws_creds: dict, db_name: str) -> str:
+    """
+    Gets DBSubnetGroup and VpcSecurityGroupId from production database needed for later requests
+
+    Args:
+        aws_creds (dict): dictionary containing aws credentials
+        db_name (str): name of production database cluster
+
+    Returns:
+        str: DBSubnetGroup and VpcSecurityGroupId from production database
+    """
+    rds = create_rds_client(aws_creds)
+
+    response = rds.describe_db_clusters(
+        DBClusterIdentifier=db_name,
+    )
+    subnet_group = response["DBClusters"][0].get("DBSubnetGroup")
+    security_group = response["DBClusters"][0]["VpcSecurityGroups"][0].get(
+        "VpcSecurityGroupId"
+    )
```

Comment:
I suppose I am banking on those parts of the response dictionary always being present as described in the boto3 docs. I can certainly add handling for `None` being returned if that is your recommendation.

Reply:
In that case, is there a reason why you aren't using regular dictionary indexing using `[]`? I'm just wondering why you switch back and forth between `[]` and `get()`.

Reply:
I don't have a particular reason for using `.get()` in this DAG. After reading a bit about this method and how it is best used, I changed to using `[]` because we have no need to handle `None` or set default values in the absence of data.